docs: RFC 0003 Standardize Github Labels

CalvinRodo opened the pull request:
See document.

CalvinRodo committed:
feat: upload standard labels draft

dj2 commented on rfcs/xxxx-standard-gh-labels.md:
```diff
@@ -0,0 +1,118 @@
+
+## Motivation
+
+Currently every github repo has to modify existing Github labels upon creation of a new repo or use the default labels. Unfortunately GH does not provide bilingual labels out of the box and so we are non-compliant with OLA (Official Languages Act) requirements. If we customize lables for a business unit (BU) we will then have to manually copy labels into every repo that is owned by that BU.
```
nit: lables -> labels

CalvinRodo replied:
fixed

dj2 commented on rfcs/xxxx-standard-gh-labels.md:
```diff
@@ -0,0 +1,118 @@
+
+See https://github.com/cds-snc/covid-alert-server-labels for an example of what is being proposed to implement this solution. 
+
+The source of truth for these labels will reside in github and what is put forth in this document is just to start the discussion. The understanding is that if this Standard is accepted that these labels will be copied into github and maintenance and discussion will continue in those repositories. 
```
What do you mean by 'copied into github'? We'll create a fork of the labels repo above and they'll be housed there? Or, a discussion?

CalvinRodo replied:
I mean a github repo will be created to manage this action, I'll clarify text

dj2 commented on rfcs/xxxx-standard-gh-labels.md:
```diff
@@ -0,0 +1,118 @@
+The source of truth for these labels will reside in github and what is put forth in this document is just to start the discussion. The understanding is that if this Standard is accepted that these labels will be copied into github and maintenance and discussion will continue in those repositories. 
+
+### Baseline Labels
+This list of labels will be standardized across the organization and will be mandatory, they will apply high level groupings and statuses and will allow for a consistent way of talking about issues across projects 
```
nit: add .

dj2 commented on rfcs/xxxx-standard-gh-labels.md:
````diff
@@ -0,0 +1,118 @@
+Each product will be allowed to apply their own unique set of labels alongside the baseline labels. 
+
+
+- ```
````
I think your ```'s are in the wrong spots. Putting 3 before as "```yaml" and 3 at the very end should syntax highlight I think

CalvinRodo replied:
Good catch the backticks were just to format nicely but formatting as yaml makes more sense.

dj2 commented on rfcs/xxxx-standard-gh-labels.md:
````diff
@@ -0,0 +1,118 @@
+  Description:
+  Colour:
+- ```
+  Name: Security | Sécurité
````
I don't think we should propose this. We probably need security issues to _not_ go through the public trackers as they become public but to follow a separate security procedure. This may make it confusing to folks who think will then think security issues should get filed through these trackers.

CalvinRodo replied:
That makes sense but at the same time we have shared some security issues in our repos but mainly before things have made it to production. 

https://github.com/cds-snc/covid-alert-server/issues?q=is%3Aissue+label%3ACCCS+is%3Aclosed+label%3Asecurity

So I think there might be a need for this but I'm open to more discussion around it.

dj2 replied:
Maybe we just need to set the description to something like 'Non-production security issues' ? Then we have the SECURITY.md files which point to where to send production security issues.

dj2 commented on rfcs/xxxx-standard-gh-labels.md:
```diff
@@ -0,0 +1,118 @@
+
+Every label is bound by the following requirement:  
+
+- A label must be bilingual in the format `en | fr`
```
nit: change must to MUST ? same with one below

dj2 commented on rfcs/xxxx-standard-gh-labels.md:
```diff
@@ -0,0 +1,118 @@
+
+### Governance
+
+Governance surrounding labels will occur in Github repos, as labels are defined as json objects and are enforced through Github actions we can use Github to apply a level of lightweight automated workflows. Github's Code owner functionality will be used to ensure 
```
final sentence cut off.

CalvinRodo replied:
Fixed, it was a sentence I meant to make shorter.

dj2 commented on rfcs/xxxx-standard-gh-labels.md:
```diff
@@ -0,0 +1,118 @@
+## Alternatives
+What other designs and processes have been considered?
+
+## Open Questions
```
Delete these sections if they aren't needed.

dj2 commented on rfcs/xxxx-standard-gh-labels.md:
```diff
@@ -0,0 +1,118 @@
+Product specific label repos will be owned by the business unit’s SRE member or the teams 
+
+## Drawbacks
+Why should we _not_ do this?
```
The one potential drawback is folks may find this constricting in that labels are top down. I believe this is mitigated because there is the escape hatch of product/project specific labels in the proposal.

CalvinRodo replied:
Yeah that was the plan with BU's being able to extend is that it allows us to have a standard view into all projects so we can start to build up some automated reporting for senior management and to address common tasks across projects.

dj2 replied:
Yup, I think listing it as a potential drawback but give the reasoning for why we don't see it being an issue is beneficial to future readers to see that we thought about it but provided our arguments against the drawback.

CalvinRodo replied:
Updated Drawbacks

dj2 commented on rfcs/xxxx-standard-gh-labels.md:
```diff
@@ -0,0 +1,118 @@
+Why should we _not_ do this?
+
+## Alternatives
+What other designs and processes have been considered?
```
The current alternative is what we have now which is some mix of default Github labels and ad-hoc project labels. The downside is getting a holistic view of all the projects and understanding their current status along with the cognitive load of switching projects requiring learning a new set of labels.

CalvinRodo replied:
Good alternative I've added it to the doc

dj2 commented on rfcs/xxxx-standard-gh-labels.md:
```diff
@@ -0,0 +1,118 @@
+---
+created-on: 2021-03-23   (date the RFC was created)
```
(Delete the stuff in ()'s I put in in as a comment should have noted to delete them, heh)

CalvinRodo replied:
will do

dj2 commented on rfcs/xxxx-standard-gh-labels.md:
```diff
@@ -0,0 +1,118 @@
+---
```
I'd say number it 0003 as that would be the next sequential one. And, even if we reject, still good to keep it numbered in some fashion I think?

CalvinRodo replied:
Works for me

CalvinRodo committed:
fix: address GH comments

dj2 reviewed (approved):
Looks good to me.

CalvinRodo commented:
I'd like to share this with the PM Community since it has impacts on them.

dj2 commented:
Yup, as was mentioned in the dev meeting today, we should probably let this sit for ~2 weeks before we land. So, I guess my approval is just, I approve, we need to wait to see if anyone disapproves during the intervening time.

wmoussa-cds commented:
https://github.com/cds-snc/docs/pull/12#issuecomment-810616058

wmoussa-cds commented on rfcs/0003-standard-gh-labels.md:
```diff
@@ -0,0 +1,110 @@
+
+The governance surrounding the creation, modification and approval of labels will depend on what label group is being acted upon. 
+
+**CDS wide labels** will be owned by the head of product management. 
```
too soon for this proposed line, I suggest we don't dive into ownership outside dev co

CalvinRodo replied:
Makes sense will update

CalvinRodo replied:
Changing to head of development.

wmoussa-cds commented on rfcs/0003-standard-gh-labels.md:
```diff
@@ -0,0 +1,110 @@
+
+**CDS wide labels** will be owned by the head of product management. 
+
+**Product Specific Labels** will be owned by the product manager, this is done by applying a .codeowner file against the labels.js file.
```
we can leave it for the product tech lead to kick this off, and mention eventually the product manager owns it

CalvinRodo replied:
Good idea, I'll update the RFC to focus on Dev community only.

wmoussa-cds commented on rfcs/0003-standard-gh-labels.md:
```diff
@@ -0,0 +1,110 @@
+
+Any employee of CDS can open a PR or Issue updating the labels, it is recommended that these are documented with a rationale for change.
+
+#### Github Actions
```
@maxneuvians what do you think?

wmoussa-cds commented on rfcs/0003-standard-gh-labels.md:
```diff
@@ -0,0 +1,110 @@
+Product specific label repos will be owned by the business unit’s SRE member or the teams 
+
+## Drawbacks
+CDS Employees are not used to things being dictated from top down so this has a potential to cause friction with employees who are used to doing what they want.
```
I don't think it is about dictating and we don't want to dictate anything. The drawback is some product will use the new convention and some not which at the end won't allow us to reach the maximum benefit of this RFC. 

We will have to sit with the teams not using the labels and understand their concerns. 

What do you think?

wmoussa-cds replied:
But I think if you offer OLA compliant labels out of the box and easy integration, this will be attractive enough

timarney commented on rfcs/0003-standard-gh-labels.md:
````diff
@@ -0,0 +1,110 @@
+- Name: Wont fix | Ne résoudra pas
+  Description:
+  Colour:
+```
````
Would suggest an a11y label be included here `baseline`

CalvinRodo replied:
Agreed.

dj2 replied:
Can we call it `accessibility` instead of `a11y`. Numbered words aren't understandable unless you have context.

CalvinRodo replied:
yeah I used the longform for it since https://mobilea11y.com/blog/inaccessible-a11y/

CalvinRodo committed:
docs: integrating feedback from PR

- focusing on dev community only
- adding accessibilty tag
- revising drawbacks to highlight a non-uniform adoption of this RFC.

CalvinRodo committed:
Merge branch 'main' into docs/standard-labels

wmoussa-cds commented:
Thanks @CalvinRodo for suggesting and sorry for the delay.

dj2 commented:
As an FYI, the CDS-SNC default repository settings are updated to use the labels presented in this RFC. Projects created after this point should have the new labels by default.

wmoussa-gc commented:
@dj2 is there a plan to migrating existing repo to this new set of labels?

dj2 commented:
@CalvinRodo is setting up a github action that will apply these labels automatically. We won't apply them to old repos without help from the dev team impacted, as I believe you need to do some work to migrate.
